enterprise/cmd/repo-updater/authz/request_queue.go

```diff
@@ -0,0 +1,205 @@
+package authz
+
+import (
+	"container/heap"
+	"sync"
+)
+
+// Priority defines how urgent the permissions syncing request is.
+// Generally, if the request is driven from a user action (e.g. sign up, log in)
+// then it should be PriorityHigh. All other cases should be PriorityLow.
+type Priority int
+
+const (
+	PriorityLow Priority = iota
+	PriorityHigh
+)
+
+// requestType is the type of the permissions syncing request. It defines the
+// permissions syncing is either repository-centric or user-centric.
+type requestType int
+
+const (
+	requestTypeRepo requestType = iota
+	requestTypeUser
+)
+
+// syncRequest is a permissions syncing request with its current status in the queue.
+type syncRequest struct {
+	typ      requestType
+	id       int32
+	priority Priority
+
+	updating bool // Whether the request has been acquired
```

Comment:
```suggestion
	acquired bool // Whether the request has been acquired
```

That would've helped me a lot reading the rest of the code

Reply:
Hmm... the truth is when this value is `true`, we _are_ updating the permissions for this request 🤔

Reply:
Yeah, but from the view of the queue, the request has been acquired. It doesn't know whether it's used to update/delete/add permissions, right?

And the comment next to this field gives a strong hint that maybe `acquired` is the better name? 😄

Reply:
Hmm, OK, I'll update!